libcaf_core/src/actor_system.cpp

```diff
@@ -135,12 +139,20 @@ actor_system::actor_system(actor_system_config&& cfg)
 
 actor_system::~actor_system() {
   // stop modules in reverse order
+  groups_.stop();
   registry_.stop();
   for (auto i = modules_.rbegin(); i != modules_.rend(); ++i)
     if (*i)
       (*i)->stop();
   logger_.stop();
   CAF_SET_LOGGER_SYS(nullptr);
+  CAF_ASSERT(num_running_actors_ == 0);
+  // calls `std::terminate()` if running actors exist here
+  // follows the convention of `std::thread` destruction
+  if (num_running_actors_ > 0) {
+    CAF_LOG_ERROR("actors running upon hosting actor system destruction");
```

Comment:
Thanks. I didn't notice this. But, fortunately, it turns out that `CAF_LOG_ERROR()` does not use the logger of the destructing actor system. See [here](https://github.com/actor-framework/actor-framework/blob/topic/actor-system/libcaf_core/caf/logger.hpp#L193) for the implementation of `CAF_LOG_ERROR()`. Macros like this and `CAF_ASSERT()` should be usable outside the context of an actor system. As a result, they do not depend on anything of it.